rules/scala/private/provider.bzl

```diff
@@ -1,16 +1,37 @@
-load("@rules_scala_annex//rules/scala:provider.bzl", "BasicScalaConfiguration", "ScalaConfiguration")
+load(
+    "@rules_scala_annex//rules/scala:provider.bzl",
+    old_BasicScalaConfiguration = "BasicScalaConfiguration",
+    old_ScalaConfiguration = "ScalaConfiguration",
+)
+load(
+    "@rules_scala_annex//rules:providers.bzl",
+    new_ScalaConfiguration = "ScalaConfiguration",
+    new_ZincConfiguration = "ZincConfiguration",
+)
 
 def annex_configure_basic_scala_implementation(ctx):
-    return [BasicScalaConfiguration(
-        version = ctx.attr.version,
-        compiler_classpath = ctx.files.compiler_classpath,
-        runtime_classpath = ctx.attr.runtime_classpath,
-    )]
+    return [
+        old_BasicScalaConfiguration(
+            version = ctx.attr.version,
+            compiler_classpath = ctx.files.compiler_classpath,
+            runtime_classpath = ctx.attr.runtime_classpath,
+        ),
+        new_ScalaConfiguration(
+            version = ctx.attr.version,
+            compiler_classpath = ctx.files.compiler_classpath,
+            runtime_classpath = ctx.attr.runtime_classpath,
+        ),
+    ]
```

Comment:
You're going to get right of old/new before merging, I assume.

Reply:
I'm going to remove this entire file in favor of some newer rules and the newer providers.